src/utils/geolocation-manager.ts

```diff
@@ -0,0 +1,51 @@
+import { Map, View, Geolocation } from 'ol';
+import GeolocationMarker from '../components/geolocation-marker';
+import LoaderBoxControl from '../components/notification/loader';
+
+export default class GeolocationManager {
+  geolocation: Geolocation;
+  marker: GeolocationMarker | undefined;
+
+  removeLoaderBox(map: Map) {
+    map.getControls().forEach((control) => {
+      if (control instanceof LoaderBoxControl) {
+        map.removeControl(control);
+      }
+    });
+  }
+
+  constructor(map: Map, view: View) {
+    this.geolocation = new Geolocation({
+      trackingOptions: {
+        enableHighAccuracy: true,
+      },
+      projection: view.getProjection(),
+    });
+
+    this.geolocation.setTracking(true);
```

Comment:
not sure we need to track the user position, probably only once is enough and they can click again if needed. Will save battery

Reply:
Regarding the firefox problem I investigate. The problem is the same with safari.